### PR TITLE
Feature/timeline details

### PR DESCRIPTION
**Pull request by kcjerrell**
(note: this PR looks bigger than it actually is because of a .css file being reformatted on save.)

# Description

[Explanation video](https://www.loom.com/share/0725dd7e16e44ec7acd7485a1b01bdb3)
[Code explanation video](https://www.loom.com/share/061df9f5083347eb899110b44599c11a)

- Updates the timeline to correctly show a title (if there is one) or the desc... if there is not.
- Adds a title field and a tags field to the CompleteIncident form.
- Adds a helper function to convert force_rank from the short form to the longer, descriptive form.

## Type of change
- [x] New feature (non-breaking change which adds functionality)

## Change Status
- [x] Complete, tested, ready to review and merge

# How Has This Been Tested?
- [x] `npm test`
- [x] new tests added for forceRankHelper 

# Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

### CODE REVIEW

**Review by austin-carman — APPROVED** [review]
Overall looks good. Maybe could have split this up into a couple/few pull requests but otherwise seems ok

**Comment by kcjerrell** [comment]
> Overall looks good. Maybe could have split this up into a couple/few pull requests but otherwise seems ok

Yeah, the timeline changes could be separated from the incident editing changes, but since I need to be able to change an incident's title to test the timeline, they ended up in the same PR.

File: src/components/AdminDashboard/CompleteIncident.js
Comment (Ethan-Edmond) [inline src/components/AdminDashboard/CompleteIncident.js]: Should this be htmlFor="title" ?

Comment (Ethan-Edmond) [reply src/components/AdminDashboard/CompleteIncident.js]: Actually looking closer, a bunch of the htmlFors don't match their respective input fields, not just this one.

File: src/components/AdminDashboard/CompleteIncident.js
Comment (Ethan-Edmond) [inline src/components/AdminDashboard/CompleteIncident.js]: If incident.tags is empty, won't incident.tags.join(', ') evaluate to '' anyways?

File: src/components/AdminDashboard/CompleteIncident.js
Comment (Ethan-Edmond) [inline src/components/AdminDashboard/CompleteIncident.js]: Later could we perhaps move the form and display elements into respective components and write this similar to (maybe with better names): 
```jsx
{!editing ? <DisplayElems/> : <Form/>}
```
instead of repeating that ternary statement?

File: src/components/AdminDashboard/CompleteIncident.js
Comment (kcjerrell) [reply src/components/AdminDashboard/CompleteIncident.js]: That's because I'm sloppy. Fixing that now... actually I'm going to remove the htmlFors and put the inputs inside the label tags.

File: src/components/AdminDashboard/CompleteIncident.js
Comment (kcjerrell) [reply src/components/AdminDashboard/CompleteIncident.js]: Yeah, I agree. I'll split that up right now.

File: src/utils/sourceListHelper.js
Comment (Ethan-Edmond) [inline src/utils/sourceListHelper.js:25]: Is there a reason to have
```jsx
!sources ||
```
in this conditional?

File: src/components/incidents/Incidents.js
Comment (Ethan-Edmond) [inline src/components/incidents/Incidents.js]: Should this be 
```jsx
incident?.src
```
?

File: src/components/timeline/TimelineDetailsExpander.js
Comment (Ethan-Edmond) [inline src/components/timeline/TimelineDetailsExpander.js]: Is the whitespace here a github thing? Because I don't think that what I'm seeing is what you meant.

File: src/components/timeline/TimelineItems.css
Comment (Ethan-Edmond) [inline src/components/timeline/TimelineItems.css:2]: Could you perhaps explain what you changed in this css file? Sorry I'm having some trouble going through it.

File: src/utils/sourceListHelper.js
Comment (kcjerrell) [reply src/utils/sourceListHelper.js:25]: I'm not sure, I didn't write it originally. I just changed the function from requiring the entire incident since it only accesses the src property. I added the isArray() check to pass a test that was failing.

File: src/components/AdminDashboard/CompleteIncident.js
Comment (kcjerrell) [reply src/components/AdminDashboard/CompleteIncident.js]: I didn't split it up into separate components as this PR is already sprawling... but I did combine the ternary statements

File: src/utils/forceRankHelper.js
Comment (Ethan-Edmond) [inline src/utils/forceRankHelper.js]: This if statement doesn't look broken, but I would recommend using braces, just for style.

File: src/components/AdminDashboard/CompleteIncident.js
Comment (Ethan-Edmond) [reply src/components/AdminDashboard/CompleteIncident.js]: Yea, I think that could be put in another PR/ticket/whatever. We can get it later.

File: src/components/timeline/TimelineItems.css
Comment (kcjerrell) [reply src/components/timeline/TimelineItems.css:2]: Yeah, the whole thing auto-formatted when I saved the changes.

 - I removed the padding from h4.cityState
 - I added h3.card-desc

File: src/utils/sourceListHelper.js
Comment (Ethan-Edmond) [reply src/utils/sourceListHelper.js:25]: It doesn't really break anything to have it, it's just redundant as far as I can tell. So don't worry about it I guess.

File: src/components/timeline/TimelineItems.css
Comment (Ethan-Edmond) [reply src/components/timeline/TimelineItems.css:2]: Thanks!

File: src/components/AdminDashboard/CompleteIncident.js
Comment (kcjerrell) [reply src/components/AdminDashboard/CompleteIncident.js]: Yeah, but not if tags was null or undefined (for some reason)

File: src/components/AdminDashboard/CompleteIncident.js
Comment (kcjerrell) [reply src/components/AdminDashboard/CompleteIncident.js]: After going through this component more thoroughly there's some weird things going on with prop-drilling, and the handleSubmit function is weird too...

File: src/components/AdminDashboard/CompleteIncident.js
Comment (Ethan-Edmond) [reply src/components/AdminDashboard/CompleteIncident.js]: Yeah I suppose null doesn't have a join method. Was that a problem you ran into?

**Comment by Ethan-Edmond** [comment]
Ok, I think I read all of it. It looks like it does what you said and a bit more and doesn't break anything. Ill approve it once the merge conflict is resolved between this and your other PR.

**Review by Ethan-Edmond — APPROVED** [review]
Cool, looks great. The biggest improvement I can suggest is to cut off the PR before it gets so big. If you can't cover it in your explanation, do it in another PR. You don't have to make your PRs have a 1 to 1 relationship with the tickets. Also, sorry about making you run around so much. I was just trying to get clarification on some things, not make you redo your whole PR. Nice job on this!

File: src/components/AdminDashboard/CompleteIncident.js
Comment (kcjerrell) [reply src/components/AdminDashboard/CompleteIncident.js]: In CompleteIncident.test.js, the incident data they are using doesn't have a tags property, and neither does any of the data in mockStoreData. That's probably something I should fix, huh?

**Review by paulstgermain — CHANGES_REQUESTED** [review]
Video Walkthrough Feedback

- Just like the last time, the way you presented this is top notch! Love your use of the current deployed version as a comparison to your current work on your local machine as a means of highlighting the changes made.
- That said, I'd love for you to spend more time with your code in these videos to balance that out! I only caught a brief snippet of your TimelineDetailsExpander component (Which is brilliant, by the way!) and you never walked us through the other changes you made in your code to implement this and the other changes you made.
- Remember: the main purpose of these videos is to give you all practice talking through the technical aspects of your day-to-day work, to better prepare you for interviews and interfacing with teams of varying experience - so let's talk code!

@kcjerrell I'd like to see you upload another walkthrough for this PR, this time focusing on the code in the PR itself, then I'll continue my review!

**Review by ahmedseragcodes — APPROVED** [review]
Changes make sense, don't seem room for bugs/error, approved

**Review by paulstgermain — APPROVED** [review]

**Pull request merged**
